docker/ubuntu20.04/Dockerfile

```diff
@@ -0,0 +1,39 @@
+FROM ubuntu:20.04
+ENV DEBIAN_FRONTEND noninteractive
+
+## Update cache and upgrade image
+RUN apt-get -y update && apt-get -y upgrade && apt-get -y dist-upgrade
```

Comment:
Seems the setup process for all distros are almost the same, perhaps it could be extracted into a script, then all Dockerfiles could just run the shared script?

Reply:
There could be subtle differences, esp. since I will add rpm and yum (CentOS, RHEL) later on. I think it also helps visually to see the sequence of commands in a central spot. One exception is the CMake-installation script, the rest of deps are all described in `Dockerfile` ... They are usually almost the same.. until they are not 😄